applications/nrf_desktop/src/modules/ble_state.c

```diff
@@ -198,21 +199,27 @@ static void exchange_func(struct bt_conn *conn, u8_t err,
 static void security_changed(struct bt_conn *conn, bt_security_t level,
 			     enum bt_security_err bt_err)
 {
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+		__ASSERT_NO_MSG(active_conn[0] == conn);
+		k_delayed_work_cancel(&security_timeout);
+	}
+
 	char addr[BT_ADDR_LE_STR_LEN];
 
 	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
 	if (!bt_err) {
+		__ASSERT_NO_MSG(level == BT_SECURITY_L2);
```

Comment:
are we sure this will always be the same security level? maybe instead we should disconnect if level is incorrect?
```
if (!bt_err && (level == L2)) {
   LOG("all is well");
} else {
   ... oopsie ^^
}

Reply:
ok